haskell-language-server.cabal

```diff
@@ -42,18 +42,13 @@ common defaults
 common test-defaults
   ghc-options:        -threaded -rtsopts -with-rtsopts=-N
 
-common common-deps
```

Comment:
Deleting this for now as it's contributing more to unused-packages warnings than it's saving duplication..
Will reintroduce this later where it makes sense.

Reply:
mostly a workaround for not yet having https://github.com/haskell/cabal/issues/9569